### PR TITLE
Fix PostIterator, if post is null

### PR DESCRIPTION
**Pull request by rubas**
Related: #2917 

## Issue
`$post` can be `null`, which breaks the code.

### CODE REVIEW

**Comment by nlemoine** [comment]
Thank you @rubas for this PR. 

We still support PHP 7.4 so `$post?->setup();` syntax will not work. Can you adapt your PR to match 7.4 requirements?

**Comment by rubas** [comment]
`protected null|Post $last_post;`  

The support for union type was only introduced in PHP 8.0.

**Comment by rubas** [comment]
The failed coding standard test is outside of this PR. I'm happy to push a fix...


```php
1) tests/test-timber-cache.php

    ---------- begin diff ----------
@@ -329,7 +329,7 @@
         $clear = $loader->clear_cache_timber(Timber\Loader::CACHE_OBJECT);
         $this->assertTrue($clear);
         $works = true;
-      
+
         if (isset($wp_object_cache->cache[Timber\Loader::CACHEGROUP])
             && !empty($wp_object_cache->cache[Timber\Loader::CACHEGROUP])) {
             $works = false;
    ----------- end diff -----------


Applied checkers:

 * PhpCsFixer\Fixer\Whitespace\NoWhitespaceInBlankLineFixer
```

**Comment by Levdbas** [comment]
Thank you @rubas  for taking this on. This might fix https://github.com/timber/timber/issues/2859 as well?

**Comment by nlemoine** [comment]
for coding standards, you just need to run `composer run cs:fix`. But you can leave the failing test file or fix it. We'll take care of it later.

**Comment by rubas** [comment]
> Thank you @rubas for taking this on. This might fix #2859 as well?

Correct. This is the same bug.

File: src/PostsIterator.php [hunk 1]
```diff
@@ -12,10 +12,10 @@
 class PostsIterator extends ArrayIterator
 {
     /**
-     * @var Post The last post that was returned by the iterator. Used
+     * @var null|Post The last post that was returned by the iterator. Used
      *                   to skip the logic in `current()`.
      */
-    protected Post $last_post;
+    protected $last_post;
```
Comment (nlemoine) [inline src/PostsIterator.php]: You still use a nullable type here.
```suggestion
    protected ?Post $last_post;
```

**Review by nlemoine — CHANGES_REQUESTED** [review]
Thanks @rubas. This is almost good for me.

Can you add the nullable type to `$last_post` and a test case for this?

**Review by nlemoine — APPROVED** [review]
This is all good, thanks @rubas!

**Pull request merged**